tests/mock_env.py

```diff
@@ -0,0 +1,210 @@
+"""
+Copyright (c) 2020 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+
+from __future__ import unicode_literals, absolute_import
+
+from flexmock import flexmock
+
+from atomic_reactor.constants import PLUGIN_BUILD_ORCHESTRATE_KEY
+from atomic_reactor.plugin import (PreBuildPluginsRunner,
+                                   BuildStepPluginsRunner,
+                                   PostBuildPluginsRunner,
+                                   PrePublishPluginsRunner,
+                                   ExitPluginsRunner)
+from atomic_reactor.inner import DockerBuildWorkflow
+from atomic_reactor.plugins.pre_reactor_config import (ReactorConfig,
+                                                       ReactorConfigPlugin,
+                                                       WORKSPACE_CONF_KEY)
+
+from tests.constants import TEST_IMAGE, MOCK_SOURCE
+from tests.stubs import StubSource, StubInsideBuilder
+
+
+class MockEnv(object):
+    """
+    Mock environment for unit tests.
+
+    Provides methods for setting up workflow (DockerBuildWorkflow) and runner (PluginsRunner)
+    for a specific test scenario.
+
+    Example usage:
+    >>> runner = (MockEnv()
+    >>>           .for_plugin('prebuild', 'my_plugin')
+    >>>           .set_scratch(True)
+    >>>           .make_orchestrator()
+    >>>           .create_runner(docker_tasker))  # docker_tasker is a fixture
+    >>> runner.run()
+    """
+
+    _plugin_phases = ('prebuild', 'buildstep', 'postbuild', 'prepublish', 'exit')
+
+    _runner_for_phase = {
+        'prebuild': PreBuildPluginsRunner,
+        'buildstep': BuildStepPluginsRunner,
+        'postbuild': PostBuildPluginsRunner,
+        'prepublish': PrePublishPluginsRunner,
+        'exit': ExitPluginsRunner,
+    }
+
+    _results_for_phase = {
+        'prebuild': 'prebuild_results',
+        'buildstep': 'buildstep_result',
+        'postbuild': 'postbuild_results',
+        'prepublish': 'prepub_results',
+        'exit': 'exit_results',
+    }
+
+    _plugins_for_phase = {phase: phase + '_plugins_conf' for phase in _plugin_phases}
+
+    def __init__(self):
+        self.workflow = DockerBuildWorkflow(TEST_IMAGE, source=MOCK_SOURCE)
+        self.workflow.source = StubSource()
+        self.workflow.builder = StubInsideBuilder().for_workflow(self.workflow)
+        self.workflow.builder.tasker = flexmock()
+
+        self._phase = None
+        self._plugin_key = None
+
+    def create_runner(self, docker_tasker):
+        """
+        Create runner for current plugin (configured using for_plugin())
+
+        :param docker_tasker: docker_tasker fixture from conftest
```

Comment:
It would be good the document the return value as well. This method is different from other MockEnv instance methods by not return itself.

Reply:
Good point! Added